### PR TITLE
Scitoken

### PR DESCRIPTION
**Pull request by LinhTangTD**
Clean Code for Auth0 integration

### CODE REVIEW

File: cashman-flask-project/cashman/index.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    incomes.clear()
+    incomes.append(request.get_json())
+    return "", 204
+
+
+@app.route("/incomes/private", methods=["DELETE"])
+@requires_auth
+def remove_income():
+    # Delete operation requires different scope's permission
+    if requires_scope("delete:income"):
```
Comment (jbasney) [inline cashman-flask-project/cashman/index.py]: How about adding a required scope for other routes?

**Review by jbasney — COMMENTED** [review]
Looks good. Please add an acknowledgement in the README.md about where the original code came from (https://github.com/auth0-blog/flask-restful-apis).

**Comment by djw8605** [comment]
Indeed, for example, the heroku scitokens plugin includes similar auth line, but with more options:
```python
@scitokens_protect.protect(audience="https://demo.scitokens.org", scope="read:/protected", issuer=["https://demo.scitokens.org", "https://cilogon.org"])
```

https://github.com/scitokens/scitokens-heroku/blob/master/scitokens_protect.py

**Comment by jbasney** [comment]
I added the Auth0 acknowledgement to README.md in 3d3beff65b3c1b25728875e0cd6f6e83aa300e70 since I wanted to also upload the swimlane diagrams that we worked on.

**Pull request closed without merging**